sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -1527,14 +1708,22 @@ static class DatastoreWriterFn extends DoFn<Mutation, Void> {
         V1DatastoreFactory datastoreFactory,
         WriteBatcher writeBatcher) {
       this.projectId = checkNotNull(projectId, "projectId");
+      this.databaseId = DEFAULT_DATABASE;
       this.localhost = localhost;
       this.datastoreFactory = datastoreFactory;
       this.writeBatcher = writeBatcher;
     }
 
     @StartBundle
     public void startBundle(StartBundleContext c) {
-      datastore = datastoreFactory.getDatastore(c.getPipelineOptions(), projectId.get(), localhost);
+      // Read Firestore DatabaseID from FirestoreOptions.
+      String firestoreDatabaseId =
```

Comment:
why are we passing the databaseId around, but querying for it from options here? I would think that it should be passed in from the constructor.

Reply:
We hope users can run the datastore beam with flag for write. StartBundleContext. 
```
  "--firestoreDb=${firestoreDb}",
```

Similar example in Firestore: 
https://github.com/apache/beam/blob/d2663f95af3698c10ba983f5b967108bd744595e/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/FirestoreV1WriteFn.java#L206

Reply:
A user could do .withDatabaseId(ID), and that doesn't appear to do anything

Reply:
I think this is an antipattern. I don't think we should be passing in options via options. What if the user wanted to query from two databases in a single pipeline?